### PR TITLE
Unit tests dwi

### PR DESCRIPTION
**Pull request by EmmaRenauld**
# Quick description

Unit tests for the dwi files.

Changes:
- Now the dwi outlier detection takes into consideration the symmetry of b-vectors.
- In compute_dwi_attenuation: instead of dividing by 0 and then removing infinite values (which generates warning) the division by 0 is handled beforehand.

## Type of change

Check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Unit tests

## Provide data, screenshots, command line to test (if relevant)

...

# Checklist

- [x] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [x] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by arnaudbore — APPROVED** [review]
GTG wait for @karanphil  for a second quick review

File: scilpy/dwi/operations.py [hunk 1]
```diff
@@ -119,104 +122,168 @@ def compute_dwi_attenuation(dwi_weights: np.ndarray, b0: np.ndarray):
...
     dwi_attenuation = dwi_weights / b0
-
-    # Make sure we didn't divide by 0.
-    dwi_attenuation[np.logical_not(np.isfinite(dwi_attenuation))] = 0.
+    dwi_attenuation *= ~zeros_mask[:, :, :, None]
 
     return dwi_attenuation
 
 
 def detect_volume_outliers(data, bvecs, bvals, std_scale,
```
Comment (karanphil) [inline scilpy/dwi/operations.py:150]: Could we change the order of the parameters here? We tend to give data, bvals, bvecs usually? (Don't forget to also change it in the docstring below)

File: scilpy/dwi/operations.py [hunk 2]
```diff
@@ -119,104 +122,168 @@ def compute_dwi_attenuation(dwi_weights: np.ndarray, b0: np.ndarray):
...
+    Detects outliers. Finds the 3 closest angular neighbors of each direction
+    (per shell) and computes the voxel-wise correlation.
+    If the angles or correlations to neighbors are below the shell average (by
+    std_scale x STD) it will flag the volume as a potential outlier.
+
     Parameters
     ----------
     data: np.ndarray
-        The dwi data.
+        The 4D dwi data.
```
Comment (karanphil) [inline scilpy/dwi/operations.py:161]: If we want to be uniform, you could use the same formatting as in reconst.frf.compute_ssst_frf for the docstring of data, bvals and bvecs.

File: scilpy/dwi/operations.py [hunk 3]
```diff
@@ -119,104 +122,168 @@ def compute_dwi_attenuation(dwi_weights: np.ndarray, b0: np.ndarray):
...
     bvecs: np.ndarray
-        The bvecs
+        The bvecs, of shape (nb_gradients, 3), normalized
```
Comment (karanphil) [inline scilpy/dwi/operations.py:163]: I would not assume the bvecs to be normalized, rather I would test it and do it like in reconst.frf.compute_ssst_frf.

**Pull request merged**